activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -330,7 +330,8 @@ def index(column_name, options = {})
       end
 
       def foreign_key(table_name, options = {}) # :nodoc:
-        foreign_keys[table_name] = options
+        foreign_keys[table_name] ||= []
```

Comment:
I think `@foreign_keys` should be an array rather than a hash because foreign keys can be added to the same table.

```diff
--- a/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb
+++ b/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb
@@ -212,7 +212,7 @@ module ActiveRecord
       def initialize(name, temporary, options, as = nil)
         @columns_hash = {}
         @indexes = {}
-        @foreign_keys = {}
+        @foreign_keys = []
         @primary_keys = nil
         @temporary = temporary
         @options = options
@@ -330,7 +330,7 @@ module ActiveRecord
       end
 
       def foreign_key(table_name, options = {}) # :nodoc:
-        foreign_keys[table_name] = options
+        foreign_keys << [table_name, options]
       end
 
```